tdxctl/src/tboot.rs

```diff
@@ -88,12 +119,41 @@ impl<'a> Setup<'a> {
             .context("Missing tproxy_url")?;
 
         let url = format!("{}/prpc", tproxy_url);
-        let client = RaClient::new_mtls(
-            url,
-            fs::read_to_string(self.resolve("/etc/tappd/ca.cert"))?,
-            fs::read_to_string(self.resolve("/etc/tappd/tls.cert"))?,
-            fs::read_to_string(self.resolve("/etc/tappd/tls.key"))?,
-        )?;
+        let ra_validator = RaValidator {
+            remote_app_id: self.app_keys.tproxy_app_id.clone(),
+        };
+        let cert_client =
+            CertRequestClient::create(&self.app_keys, self.local_config.pccs_url.as_deref())
+                .await
+                .context("Failed to create cert client")?;
+        let client_key =
+            KeyPair::generate_for(&PKCS_ECDSA_P256_SHA256).context("Failed to generate key")?;
+        let config = CertConfig {
+            org_name: None,
+            subject: "tappd".to_string(),
+            subject_alt_names: vec![],
+            usage_server_auth: false,
+            usage_client_auth: true,
+            ext_quote: true,
+        };
+        let client_certs = cert_client
+            .request_cert(&client_key, config)
+            .await
+            .context("Failed to request cert")?;
+        let client_cert = client_certs.join("\n");
+        let ca_cert = self.app_keys.ca_cert.clone();
+        let client = RaClientConfig::builder()
+            .remote_uri(url)
+            .maybe_pccs_url(self.local_config.pccs_url.clone())
+            .tls_client_cert(client_cert)
+            .tls_client_key(client_key.serialize_pem())
+            .tls_ca_cert(ca_cert)
+            .tls_built_in_root_certs(false)
+            .tls_no_check(self.app_keys.tproxy_app_id == "any")
+            .cert_validator(Box::new(move |cert| ra_validator.validate(cert)))
+            .build()
+            .into_client()
+            .context("Failed to create RA client")?;
```

Comment:
Review Hint: Connect to tproxy here. If the KMS(smart contract) allow to connect to any tproxy, then we don't validate remote cert, otherwise, do bidirectional ra verification, and ensure the remote app-id matches the one KMS returned.